exo/main.py

```diff
@@ -76,6 +78,8 @@
 
 if args.discovery_module == "udp":
   discovery = UDPDiscovery(args.node_id, args.node_port, args.listen_port, args.broadcast_port, lambda peer_id, address, device_capabilities: GRPCPeerHandle(peer_id, address, device_capabilities), discovery_timeout=args.discovery_timeout)
+elif args.discovery_module == "manual":
+  discovery = ManualDiscovery(discovery_config=args.discovery_config, discovery_timeout=args.discovery_timeout)
```

Comment:
You need to pass in a `create_peer_handle` function like above with `UDPDiscovery`

Reply:
elif args.discovery_module == "manual":
  discovery = ManualDiscovery(lambda peer_id, address, device_capabilities: GRPCPeerHandle(peer_id, address, device_capabilities), discovery_config=args.discovery_config, discovery_timeout=args.discovery_timeout)

Reply:
I have created the handle, but frankly, I wanted to read the config file and define the handle from it.
If I don't do that inside the class ....

Reply:
Hello, just put the read of topology in ReadManualConfig class.